lib/scripts/createGenesisDao.ts

```diff
@@ -23,135 +29,132 @@ interface FounderSpec {
+    const UController = await Utils.requireContract("UController");
+    const universalControllerInst = await UController.deployed();
 
-    console.log(`Deploying to ${network}, gasLimit: ${gasAmount}, ${founders.length} founders`);
+    const internalFoundersConfigLocation = "../../migrations/founders.json";
+    const foundersConfig = require(internalFoundersConfigLocation).founders;
 
-    await deployer.deploy(ControllerCreator, { gas: gasAmount });
-    const controllerCreator = await ControllerCreator.deployed();
-    await deployer.deploy(DaoCreator, controllerCreator.address, { gas: gasAmount });
-    const daoCreatorInst = await DaoCreator.deployed(controllerCreator.address);
+    const customFoundersConfigLocation = foundersConfigurationLocation || internalFoundersConfigLocation;
+
+    if (internalFoundersConfigLocation !== customFoundersConfigLocation) {
+      console.log(`merging custom founders from ${customFoundersConfigLocation}`);
+      const customFoundersConfig = require(customFoundersConfigLocation).founders;
+      // merge the two
+      Object.assign(foundersConfig, customFoundersConfig);
+    }
+
+    const founders = foundersConfig[this.network];
+
+    if (!founders || (founders.length === 0)) {
+      throw new Error(`no founders were given for the network: ${this.network}`);
+    }
+
+    let gasLimit = computeForgeOrgGasLimit(founders.length);
+    const maxGasLimit = await computeMaxGasLimit(this.web3);
+
+    gasLimit = Math.min(gasLimit, maxGasLimit);
+
+    console.log(`Forging ${orgName} to ${this.network}, gasLimit: ${gasLimit} and ${founders.length} founders...`);
 
-    await deployer.deploy(UController, { gas: gasAmount });
-    const universalControllerInst = await UController.deployed();
     /**
      * Create the Genesis DAO
      */
-    const tx = await daoCreatorInst.forgeOrg(
+    const txForgeOrg = await daoCreatorInst.forgeOrg(
       orgName,
       tokenName,
       tokenSymbol,
       founders.map((f: FounderSpec) => f.address),
-      founders.map((f: FounderSpec) => web3.toWei(f.tokens)),
-      founders.map((f: FounderSpec) => web3.toWei(f.reputation)),
+      founders.map((f: FounderSpec) => this.web3.toWei(f.tokens)),
+      founders.map((f: FounderSpec) => this.web3.toWei(f.reputation)),
       universalControllerInst.address,
-      web3.toWei(100000000), // token cap of one hundred million GEN, in Wei
-      { gas: gasAmount });
+      this.web3.toWei(100000000), // token cap of one hundred million GEN, in Wei
+      { gas: gasLimit });
 
-    // maximize chances that Avatar.at will succeed
-    if (network === "live") {
-      await UtilsInternal.sleep(60000);
+    let avatarInst;
+
+    /**
+     * save info for later steps
+     */
+    while (!avatarInst) {
+      avatarInst = await Avatar.at(txForgeOrg.logs[0].args._avatar);
+      console.log("sleeping until Avatar is mined...");
+      UtilsInternal.sleep(10000); // sleep and retry until avatarInst is ready
```

Comment:
I guess that 10 seconds sleep is too much here..
You can sleep for 1 second.
And add a comment that it is "not nice hack".